07_boolean_2/main.go

```diff
@@ -15,56 +15,136 @@ E.g.:
 */
 
 func main() {
-	fmt.Println(false && (false || true))
-	fmt.Println((false && false) || true)
+	// fmt.Println(false && (false || true))
+	// fmt.Println((false && false) || true)
+	logicalTable(true, true)
+	logicalTable(true, false)
+	logicalTable(false, false)
+	logicalTable(false, true)
+	strIntCheck("", 0)
+	strIntCheck("Gopher", 3)
+	strIntCheck("Git", 2)
+	strIntCheck("Go", 4)
+}
 
-	// TODO: Write a function that prints a truth table for all the following
-	// Operations:
-	//  - AND
-	//  - OR
-	//  - NAND (NOT AND)
-	//  - NOR  (NOT OR)
-	//  - XOR  (EXCLUSIVE OR)
-	// E.g.: logicalTable(true, true) would print:
-	// AND : true
-	// OR  : true
-	// NAND: false
-	// ...
-	// Call this function for all 4 combinations of true and false.
+// TODO: Write a function that prints a truth table for all the following
+// Operations:
+//   - AND
+//   - OR
+//   - NAND (NOT AND)
+//   - NOR  (NOT OR)
+//   - XOR  (EXCLUSIVE OR)
+//
+// E.g.: logicalTable(true, true) would print:
+// AND : true
+// OR  : true
+// NAND: false
+// ...
+// Call this function for all 4 combinations of true and false.
+func logicalTable(bool1 bool, bool2 bool) {
+	// Varible used
+	// var AND bool = bool1 && bool2
+	// fmt.Println(AND)
+	// Hardcoded print results
+	// if bool1 == true && bool2 == true {
+	// 	fmpt.Println("true")
+	// }
+	// Direct print statements
+	// AND
+	fmt.Println("AND", bool1 && bool2)
+	// OR
+	fmt.Println("OR", bool1 || bool2)
+	// NAND
+	if bool1 == true && bool2 == true {
+		fmt.Println("NAND", false)
+	} else {
+		fmt.Println("NAND", true)
+	}
+	//  NOR
+	if bool1 == false && bool2 == false {
+		fmt.Println("NOR", true)
+	} else {
+		fmt.Println("NOR", false)
+	}
+	// XOR
+	if bool1 == true && bool2 == false || bool2 == true && bool1 {
+		fmt.Println("XOR", true)
+	} else if bool1 == bool2 || bool2 == bool1 {
+		fmt.Println("XOR", false)
+	}
```

Comment:
No, use logical operators.

```suggestion
  fmt.Println("NAND", ???)
	fmt.Println("NOR", ???)
	// if you really want to do an extra step, use a variable.
  var xor bool = ???
	fmt.Println("XOR", xor)
```

Reply:
Done.